### PR TITLE
Fixed bug in ELC

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
Fixes ELC part of #1559 and probalby #229  

 - Found small but fatal bug in elc.cpp. Also I could no longer reproduce the error in ELCIC and it agrees with MMM2D with two particles and metallic plates.

### CODE REVIEW

**Review by KaiSzuttor — COMMENTED** [review]
please remove debug printfs

File: src/python/espressomd/particle_data.pyx [hunk 1]
```diff
@@ -165,7 +165,9 @@ cdef class ParticleHandle(object):
                 ppos[i] = self.particle_data.get()[0].r.p[i]
 
             unfold_position(ppos, img)
-            return np.array([ppos[0], ppos[1], ppos[2]])
+            ret = np.array([ppos[0], ppos[1], ppos[2]])
```
Comment (fweik) [inline src/python/espressomd/particle_data.pyx]: This should not be in here.

File: src/core/elc.cpp [hunk 2]
```diff
@@ -1316,6 +1327,7 @@ int ELC_set_params(double maxPWerror, double gap_size, double far_cut,
     elc_params.far_calculated = 0;
   } else {
     elc_params.far_calculated = 1;
+    printf("ELC_set_params tune\n");
```
Comment (fweik) [inline src/core/elc.cpp]: Left over printf

File: src/core/elc.cpp [hunk 3]
```diff
@@ -1230,14 +1233,22 @@ void ELC_init() {
...
+    p3m.params.additional_mesh[0] = 0;
+    p3m.params.additional_mesh[1] = 0;
     p3m.params.additional_mesh[2] = elc_params.space_layer;
+  } else {
+    p3m.params.additional_mesh[0] = 0;
+    p3m.params.additional_mesh[1] = 0;
+    p3m.params.additional_mesh[2] = 0;
   }
+  
+  printf("far_cut %f\n far_cut2 %f\n h %f\n space_layer %f\n space_box %f\n far_calculated %d\n addmesh %f %f %f\n",elc_params.far_cut,elc_params.far_cut2, elc_params.h, elc_params.space_layer, elc_params.space_box, elc_params.far_calculated, p3m.params.additional_mesh[0],p3m.params.additional_mesh[1],p3m.params.additional_mesh[2]);
```
Comment (fweik) [inline src/core/elc.cpp]: Left over printf

File: src/core/elc.cpp [hunk 4]
```diff
@@ -1230,14 +1233,22 @@ void ELC_init() {
 
   if (elc_params.far_calculated && (coulomb.method == COULOMB_ELC_P3M &&
                                     elc_params.dielectric_contrast_on)) {
+    printf("ELC_init tune\n");
```
Comment (fweik) [inline src/core/elc.cpp]: Left over printf

File: src/core/elc.cpp [hunk 5]
```diff
@@ -1196,6 +1198,7 @@ int ELC_sanity_checks() {
 }
 
 void ELC_init() {
+  printf("ELC_init\n");
```
Comment (fweik) [inline src/core/elc.cpp]: Left over printf

File: src/core/elc.cpp [hunk 6]
```diff
@@ -1137,6 +1137,7 @@ double ELC_energy() {
 }
 
 int ELC_tune(double error) {
+  printf("ELC_tune\n");
```
Comment (fweik) [inline src/core/elc.cpp]: Left over printf

**Review by fweik — CHANGES_REQUESTED** [review]
Looks good to me, aside from the obvious stuff that needs to be changed.

**Review by fweik — APPROVED** [review]

**Review by fweik — APPROVED** [review]

**Pull request merged**